### PR TITLE
[GEOS-8017] Stop using a global cached schema in WFS 1.1.0 for encoding GML3 responses

### PR DESCRIPTION
**Pull request by nmco**
This PR makes WFS 1.1.0 stop using a global cached schema for all the feature types but instead build a specific schema for each request using only the necessary feature types, this is similar to what is done in WFS 2.0.0.

This PR also a test case for the situation described int he associated issues.

Please check the associated mailing list discussion for performance and CITE compliance tests.

Associated issues:
- https://osgeo-org.atlassian.net/browse/GEOS-8017 
- https://osgeo-org.atlassian.net/browse/GEOS-4773
- https://osgeo-org.atlassian.net/browse/GEOS-6014

Associated mailing list discussion: http://osgeo-org.1560.x6.nabble.com/Stop-using-a-global-cached-schema-for-all-features-types-in-WFS-1-1-0-for-encoding-GML3-responses-td5310783.html

### CODE REVIEW

**Comment by bencaradocdavies** [comment]
@nmco Travis CI reports [many app-schema unit test failures](https://s3.amazonaws.com/archive.travis-ci.org/jobs/207942836/log.txt). Some schema location failures, but many other exceptions as well.

**Comment by nmco** [comment]
The current fail is a valid fail, I'm investigating it.

**Comment by bencaradocdavies** [comment]
I tried a test local merge and a full build and app-schema online tests both passed.

**Comment by bencaradocdavies** [comment]
I have no objection to this PR being merged on master once the feature freeze is lifted.

File: src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    
+    @Override
+    public String getSchemaLocation() {
+        StringBuilder sb = new StringBuilder();
+        for (Set<FeatureTypeInfo> fts : featureTypes.values()) {
+            for (FeatureTypeInfo ft : fts) {
+                sb.append(ft.getPrefixedName()).append(",");
+            }
+        }
+        sb.setLength(sb.length()-1);
```
Comment (bencaradocdavies) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java]: Should be spaces around the "-" operator.

File: src/wfs/src/test/java/org/geoserver/wfs/v1_1/GetFeatureTest.java [hunk 2]
```diff
@@ -637,4 +641,36 @@ public void testFeatureMembers() throws Exception {
...
+    /**
+     * Test GetFeature operation invocations using virtual services end points.
+     * A common issue was XSD caches created only with the types of a certain
+     * workspace and reuse the same cache for other workspaces. The principal
+     * consequence was NULL namespaces in the GetFeature result, e.g null:Buildings.
+     */
+    @Test
+    public void testVirtualServicesInvocation() throws Exception {
+        XpathEngine xpath = XMLUnit.newXpathEngine();
+        // get ride of the current WFS schema
```
Comment (bencaradocdavies) [inline src/wfs/src/test/java/org/geoserver/wfs/v1_1/GetFeatureTest.java]: Typo: "ride" -> "rid"

@nmco your comments and English are great! I would only bother fixing this one if you find yourself updating something more substantial in this file.

File: src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java (hunk 1 above)
Comment (nmco) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java]: Good catch !

File: src/wfs/src/test/java/org/geoserver/wfs/v1_1/GetFeatureTest.java (hunk 2 above)
Comment (nmco) [inline src/wfs/src/test/java/org/geoserver/wfs/v1_1/GetFeatureTest.java]: Thanks for pointing this, I will amend it :smiley:

File: src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java (hunk 1 above)
Comment (bencaradocdavies) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/ApplicationSchemaXSD1.java]: Oh, it is the tiniest thing. Just helps prove I actually read your PR.  :-)

**Comment by aaime** [comment]
Master is ready for fresh commits, merging and let's see how CITE tests like this one.

**Pull request merged**